pkg/inclusion/paths_test.go

```diff
@@ -232,18 +245,65 @@ func Test_genSubTreeRootPath(t *testing.T) {
 	type test struct {
 		depth    int
 		pos      uint
-		expected []bool
+		expected []WalkInstruction
 	}
 	tests := []test{
-		{2, 0, []bool{false, false}},
-		{0, 0, []bool{}},
-		{3, 0, []bool{false, false, false}},
-		{3, 1, []bool{false, false, true}},
-		{3, 2, []bool{false, true, false}},
-		{5, 16, []bool{true, false, false, false, false}},
+		{2, 0, []WalkInstruction{WalkLeft, WalkLeft}},
+		{0, 0, []WalkInstruction{}},
+		{3, 0, []WalkInstruction{WalkLeft, WalkLeft, WalkLeft}},
+		{3, 1, []WalkInstruction{WalkLeft, WalkLeft, WalkRight}},
+		{3, 2, []WalkInstruction{WalkLeft, WalkRight, WalkLeft}},
+		{5, 16, []WalkInstruction{WalkRight, WalkLeft, WalkLeft, WalkLeft, WalkLeft}},
 	}
 	for _, tt := range tests {
 		path := genSubTreeRootPath(tt.depth, tt.pos)
 		assert.Equal(t, tt.expected, path)
 	}
 }
+
+func Test_calculateCommitPaths(t *testing.T) {
+	type test struct {
+		size, start, msgLen int
+		expected            []path
+	}
+	tests := []test{
+		{2, 0, 1, []path{{instructions: []WalkInstruction{WalkLeft}, row: 0}}},
+		{2, 2, 2, []path{{instructions: []WalkInstruction{}, row: 1}}},
+		{2, 1, 2, []path{{instructions: []WalkInstruction{}, row: 1}}},
+		{4, 2, 2, []path{{instructions: []WalkInstruction{WalkRight}, row: 0}}},
```

Comment:
[no change needed] visual

```ascii
row 0 -> | | |2|2|
row 2 -> | | | | |
row 3 -> | | | | |
row 4 -> | | | | |
```

merkle tree of row 0
```ascii
        x
       / \
      x   y
         / \
        2   2
```
where `y` is what we want

Reply:
exactly!